### DISCUSSION TITLE
OTA update: No interrupt since ESPhome 1.9.4?

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#47] [Q&A] [answered]
I forgot to disconnect VistaAlarm (turned it off) before updating the firmware compiled under the newly release ESPhome 1.9.4 to me ESP32 but it went through with no issue. Did something change in the OTA update process? Should we still disconnect VistaAlarm before an upload like mentioned in the Readme?

**Comment by Dilbert66** [comment, marked as answer]
It all depends on how busy the system is but it does seem to handle ota updates better since I've optimized some of the interrupt code.     You can safely update via ota without disconnecting  but there might be occasions where the update will fail so in those cases, just disconnect and try again.   The issue is more prevalent with my DSC panel version since there is more going on with that system.

**Reply by AlexandreUSA** [reply]
Thanks for the clarification.
